homework1/task04.py

```diff
@@ -0,0 +1,22 @@
+"""
+Classic task, a kind of walnut for you
+
+Given four lists A, B, C, D of integer values,
+    compute how many tuples (i, j, k, l) there are such that A[i] + B[j] + C[k] + D[l] is zero.
+
+We guarantee, that all A, B, C, D have same length of N where 0 ≤ N ≤ 1000.
+"""
+from typing import List
+
+
+def check_sum_of_four(a: List[int], b: List[int], c: List[int], d: List[int]) -> int:
+    counter = 0
+
+    for i in a:
+        for j in b:
+            for k in c:
+                for l in d:
+                    if i + j + k + l == 0:
+                        counter += 1
```

Comment:
What if you cannot use `if` here?

Reply:
Hmm, yes, I can write something like this
```Python
counter += not bool(i + j + k + l)
```
I will rewrite it

Reply:
Done

Reply:
That is one way to do it, but not that readable. Something like `counter += (i + j + k + l == 0)` is better. And the best way to solve this problem is to use `itertools`.

Reply:
Make with using itertools